src/charm.py

```diff
@@ -442,6 +444,10 @@ def _config_changed(self, event: HookEvent) -> None:
             self.model.unit.status = ActiveStatus()
 
     def _reload_configuration(self) -> None:
+        # mypy has some trouble with dynamic attributes
+        if not self._stored.setup_completed:  # type: ignore
+            logger.info("Defer starting the discourse server until discourse setup completed")
+            return
```

Comment:
Out of curiosity, is the event truly deferred? My understanding is that event.defer() needs to be called for it to be "re-queued". If it is not deferred, what triggers the supposedly deferred event to run again?

Reply:
The event is not deferred, but the `_reload_configuration` function will run in multiple hooks for example `config-changed`, `discourse-pebble-ready`, `database-relateion-change`. One of them will start the discourse server eventually.